eglib/src/gdir-unity.c

```diff
@@ -0,0 +1,133 @@
+#include <glib.h>
+
+#include "Directory-c-api.h"
+#include "Error-c-api.h"
+
+struct _GDir {
+    UnityPalFindHandle* handle;
+    gchar* current;
+    gchar* next;
+    const gchar* path_for_rewind;
+};
+
+static gboolean
+setup_dir_handle(GDir*dir, const gchar* path, GError **error)
+{
+    gchar* path_search;
+    char* result_file_name = NULL;
+    gint unused_attributes;
+    UnityPalErrorCode result;
+
+    dir->path_for_rewind = g_strdup (path);
+    path_search = g_malloc ((strlen(path) + 3)*sizeof(gchar));
+    strcpy (path_search, path);
+#ifdef G_OS_WIN32
```

Comment:
I don't think that these checks will. In a few places we need to deal with path separator differences - this is one of them. We could refactor this out into some common code. But I would rather wait to find similar usages to do that.